gradle-guide-error-prone/src/main/java/com/palantir/gradle/guide/errorprone/GetProjectInvocations.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.guide.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.google.errorprone.suppliers.Supplier;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.util.TreeScanner;
+import com.sun.tools.javac.code.Symbol.ClassSymbol;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.code.Type.ClassType;
+import java.util.List;
+
+@AutoService(BugChecker.class)
+@BugPattern(severity = SeverityLevel.ERROR, summary = GetProjectInvocations.SUMMARY)
+public final class GetProjectInvocations extends GradleGuideBugChecker implements BugChecker.MethodTreeMatcher {
```

Comment:
Not convinced about `GetProjectInvocations` as a name. I would call it `GetProjectDuringTaskExecution`. But there there are also other methods we care about people not calling during task execution (`getDependsOn()`, `getTaskDependencies()`, `getConvention()`, `getExtensions()`) which will likely add in a later PR. So because this is quite a bit more general (I doubt we'd make extra checks for these), maybe it should be something like `IllegalMethodDuringTaskExecution`?

Reply:
How does `ConfigurationCacheViolation` sound? It's direct and points people towards learning about [what is a violation[(https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:requirements). 

Or maybe `ExecutionPhaseViolation`?

Reply:
I think `ConfigurationCacheViolation` is a bit overbroad. There are other configuration cache issues (like taking an `@Input Property<Configuration>`) which this check is not going to apply to. I think `ExecutionPhaseViolation` is also a bit vague. Let's go with `IllegalMethodCalledDuringTaskExecution`? I feel it's long but at least it's accurate.

Reply:
Yep, let's go with `IllegalMethodCalledDuringTaskExecution` then.